stetho/src/main/java/com/facebook/stetho/common/android/HandlerUtil.java

```diff
@@ -62,4 +130,56 @@ public void join() {
       }
     }
   }
+
+  private static final class WaitableCallable<V> extends AbstractWaitableRunnable {
```

Comment:
I was actually about to lean in the other direction, by making a new base class that the caller could use in order to *save* an allocation. Let's call it WaitableCallable<V> and it'll implement Callable<V>, Runnable, and the code for join() / getResult().